java-checks/src/main/java/org/sonar/java/checks/AbstractOneExpectedExceptionRule.java

```diff
@@ -128,66 +127,42 @@ private static boolean isTryCatchFail(TryStatementTree tree) {
     return false;
   }
 
-  private void reportMultipleCallThrowingExceptionInTree(List<Type> expectedExceptions, Tree treeToVisit, Tree reportLocation, String placeToRefactor) {
-    List<Type> checkedTypes = expectedExceptions.stream()
-      .filter(OneExpectedCheckExceptionCheck::isChecked)
-      .collect(Collectors.toList());
-
-    if (checkedTypes.isEmpty()) {
-      return;
-    }
-
-    MethodInvocationThrowing visitor = new MethodInvocationThrowing(checkedTypes);
-    treeToVisit.accept(visitor);
-    List<Tree> invocationTree = visitor.invocationTree;
-    if (invocationTree.size() > 1) {
-      reportIssue(reportLocation,
-        String.format("Refactor the %s in order to have only one invocation throwing an expected exception.", placeToRefactor),
-        secondaryLocations(invocationTree),
-        null);
-    }
-  }
+  abstract void reportMultipleCallInTree(List<Type> expectedExceptions, Tree treeToVisit, Tree reportLocation, String placeToRefactor);
 
-  private static boolean isChecked(Type type) {
+  static boolean isChecked(Type type) {
     return !type.isSubtypeOf("java.lang.RuntimeException") && !type.isSubtypeOf("java.lang.Error");
   }
 
-  private static List<JavaFileScannerContext.Location> secondaryLocations(List<Tree> methodInvocationTrees) {
+  static List<JavaFileScannerContext.Location> secondaryLocations(List<Tree> methodInvocationTrees) {
     return methodInvocationTrees.stream()
       .map(expr -> new JavaFileScannerContext.Location("This call can throw an expected exception", expr))
```

Comment:
I really don't like this message in fact... "expected" does not make sense. I would prefer to have `"This call can throw an exception"` instead, and ideally, even the name of the exception!